tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -0,0 +1,201 @@
+IotHttpsResponseHandle_t newIotResponseHandle() {
+  IotHttpsResponseHandle_t pResponseHandle = safeMalloc(sizeof(_resHandle_t));
+  if(pResponseHandle) {
+    pResponseHandle->pBody = malloc(sizeof(uint32_t));
+    pResponseHandle->pHttpsConnection = newIotConnectionHandle();
+    pResponseHandle->pHttpsRequest = safeMalloc(sizeof(_reqHandle_t));
+    if(pResponseHandle->pHttpsRequest) {
+      pResponseHandle->pHttpsRequest->pHttpsResponse = pResponseHandle;
+    }
+    pResponseHandle->pHeaders = (uint8_t *)pResponseHandle;
+    pResponseHandle->pHeadersCur = pResponseHandle->pHeaders;
+    pResponseHandle->httpParserInfo.readHeaderParser.data = pResponseHandle; //???
+    pResponseHandle->httpParserInfo.parseFunc = http_parser_execute;
+    pResponseHandle->pReadHeaderValue = safeMalloc(sizeof(uint32_t));
+    pResponseHandle->pHeadersEnd = pResponseHandle->pHeaders + sizeof(_resHandle_t) ;
+    __CPROVER_assume(pResponseHandle->readHeaderValueLength >= 0 &&
+    pResponseHandle->readHeaderValueLength <= (pResponseHandle->pHeadersEnd - pResponseHandle->pHeaders));
+    pResponseHandle->pReadHeaderValue = malloc(pResponseHandle->readHeaderValueLength);
+  }
+  return pResponseHandle;
+}
+
+/* Creates a Request Handle and assigns memory accordingly. */
+IotHttpsRequestHandle_t newIotRequestHandle() {
+  IotHttpsRequestHandle_t requestHandle = safeMalloc(sizeof(_reqHandle_t));
+  if (requestHandle) {
+    requestHandle->pHttpsResponse = newIotResponseHandle();
+    requestHandle->pHttpsConnection = newIotConnectionHandle();
+    requestHandle->pBody = malloc(sizeof(uint32_t));
+    requestHandle->pHeaders = (uint8_t *)requestHandle;
+    requestHandle->pHeadersCur = requestHandle->pHeaders;
+    requestHandle->pHeadersEnd = requestHandle->pHeaders  + sizeof(_resHandle_t) ;
+    requestHandle->pConnInfo = newConnectionInfo();
+  }
+  return requestHandle;
+}
+
+/* Creates a Request Info and assigns memory accordingly. */
+IotHttpsRequestInfo_t * newIotRequestInfo() {
+  IotHttpsRequestInfo_t * pReqInfo = safeMalloc(sizeof(IotHttpsRequestInfo_t));
+  uint32_t bufferSize;
+  if(pReqInfo) {
+    __CPROVER_assume(bufferSize >=0 && bufferSize <= requestUserBufferMinimumSize);
+    pReqInfo->userBuffer.bufferLen = bufferSize;
+    pReqInfo->userBuffer.pBuffer = safeMalloc(bufferSize);
+  }
+  return pReqInfo;
+}
+
+/* Creates a Response Info and assigns memory accordingly. */
+IotHttpsResponseInfo_t * newIotResponseInfo() {
+  IotHttpsResponseInfo_t * pRespInfo = safeMalloc(sizeof(IotHttpsResponseInfo_t));
+  uint32_t bufferSize;
+  uint32_t bodySize;
+  if(pRespInfo) {
+    __CPROVER_assume(bufferSize >= 0 && bufferSize <= responseUserBufferMinimumSize);
+    pRespInfo->userBuffer.bufferLen = bufferSize;
+    pRespInfo->userBuffer.pBuffer = safeMalloc(bufferSize);
+    pRespInfo->pSyncInfo = malloc(sizeof(IotHttpsSyncInfo_t));
+    __CPROVER_assume(bodySize >= responseUserBufferMinimumSize);
```

Comment:
This is probably too big. The response body can be as large or small as we want. We can probably change this to
```
pRespInfo->pSyncInfo->pBody = safeMalloc(sizeof(uint32_t)) 
pRespInfo->pSyncInfo->bodyLen = sizeof(uint32_t);
```
to save memory.

Reply:
Changed to uint32_t.